api/v1alpha1/envoygateway_helpers.go

```diff
@@ -39,6 +41,11 @@ func (e *EnvoyGateway) SetEnvoyGatewayDefaults() {
 	if e.Provider == nil {
 		e.Provider = DefaultEnvoyGatewayProvider()
 	}
+	if e.Provider.Kubernetes == nil {
```

Comment:
is this needed ?
aren't you performing the same thing again inside `DefaultGateway` ?

Reply:
The DeafultGateway is overridden by the envoy cfg, this helps to ensure the defaults remain.
Incase we don't do it we need to set a default value in the envoy confg (cfgPath), which we did not want to?

```
	eg, err := config.Decode(cfgPath)
		if err != nil {
			logger.Error(err, "failed to decode config file", "name", cfgPath)
			return nil, err
		}
		// Set defaults for unset fields
		eg.SetEnvoyGatewayDefaults()
		cfg.EnvoyGateway = eg
```